Kernel/Storage/SD/Registers.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Endian.h>
+#include <AK/Types.h>
+
+namespace Kernel::SD {
+
+enum class HostVersion : u8 {
+    Version1,
+    Version2,
+    Version3,
+    Unknown
+};
+
+// SD Host Controller Simplified Specification Version 3.00
+// NOTE: The registers must be 32 bits, because of a quirk in the RPI.
+struct SDRegisters {
+    u32 argument_2;
+    u32 block_size_and_block_count;
+    u32 argument_1;
+    u32 transfer_mode_and_command;
+    u32 response_0;
+    u32 response_1;
+    u32 response_2;
+    u32 response_3;
+    u32 buffer_data_port;
+    u32 present_state;
+    u32 host_configuration_0;
+    u32 host_configuration_1;
+    u32 interrupt_status;
+    u32 interrupt_status_enable;
+    u32 interrupt_signal_enable;
+    u32 host_configuration_2;
+    u32 capabilities_0;
+    u32 capabilities_1;
+    u32 maximum_current_capabilities;
+    u32 maximum_current_capabilities_reserved;
+    u32 force_event_for_auto_cmd_error_status;
+    u32 adma_error_status;
+    u32 adma_system_address[2];
+    u32 preset_value[4];
+    u32 reserved_0[28];
+    u32 shared_bus_control;
+    u32 reserved_1[6];
+    struct [[gnu::packed]] {
+        u8 interrupt_signal_for_each_slot;
+        u8 : 8;
+        HostVersion specification_version_number;
+        u8 vendor_version_number;
+    } slot_interrupt_status_and_version;
+};
+static_assert(AssertSize<SDRegisters, 256>());
+
+union OperatingConditionRegister {
+    u32 raw;
+    struct {
+        u32 : 15;
+        u32 vdd_voltage_window_27_28 : 1;
```

Comment:
boolean for `: 1` could be easier to access… (not sure how relevant is this exactly…)

Reply:
I disagree
see <https://github.com/SerenityOS/serenity/pull/17887#discussion_r1140880456>